sound/soc/sof/ipc4-topology.c

```diff
@@ -533,7 +533,7 @@ static void sof_ipc4_widget_free_comp_dai(struct snd_sof_widget *swidget)
 	kfree(available_fmt->dma_buffer_size);
 	kfree(available_fmt->base_config);
 	kfree(available_fmt->out_audio_fmt);
-	if (ipc4_copier->dai_type == SOF_DAI_INTEL_HDA)
+	if (ipc4_copier->dai_type != SOF_DAI_INTEL_SSP && ipc4_copier->dai_type != SOF_DAI_INTEL_DMIC)
```

Comment:
SOF_DAI_INTEL_SSP = 1, SOF_DAI_INTEL_DMIC = 2,how about " ipc4_copier->dai_type > SOF_DAI_INTEL_DMIC"

Reply:
I think !SSP and !DMIC is better

Reply:
@RanderWang We need to specify SSP and DMIC. So let's leave it as it is.